### PR TITLE
CW-512 wallet sort

### PR DESCRIPTION
**Pull request by fossephate**
Issue Number (if Applicable): Fixes #

# Description

Please include a summary of the changes and which issue is fixed / feature is added. 

# Pull Request - Checklist  

- [ ] Initial Manual Tests Passed
- [ ] Double check modified code and verify it with the feature/task requirements
- [ ] Format code
- [ ] Look for code duplication
- [ ] Clear naming for variables and methods

### CODE REVIEW

File: lib/src/screens/wallet_list/filtered_list.dart [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+class FilteredListState extends State<FilteredList> {
+  @override
+  Widget build(BuildContext context) {
+    return Observer(
+      builder: (_) => ReorderableListView.builder(
+        physics: const BouncingScrollPhysics(),
+        itemBuilder: widget.itemBuilder,
+        itemCount: widget.list.length,
+        onReorder: (int oldIndex, int newIndex) {
+          setState(() {
```
Comment (OmarHatem28) [inline lib/src/screens/wallet_list/filtered_list.dart:29]: wouldn't this cause a duplicate rebuild? wouldn't the observer already rebuild it when the observable list changes?

File: lib/entities/wallet_list_order_types.dart [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  String toString() {
+    switch (this) {
+      case WalletListOrderType.CreationDate:
+        return S.current.creation_date;
+      case WalletListOrderType.Alphabetical:
+        return S.current.alphabetical;
+      case WalletListOrderType.GroupByType:
+        return S.current.group_by_type;
+      case WalletListOrderType.Custom:
+        return S.current.custom;
```
Comment (OmarHatem28) [inline lib/entities/wallet_list_order_types.dart:19]: let's change this to be `Custom (Hold and Drag)` or if you have a better wording that elaborates the custom sorting mechanism

File: lib/src/screens/dashboard/widgets/filter_list_widget.dart [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+                    WalletListOrderType.CreationDate.toString(),
+                    style: TextStyle(
+                        color: Theme.of(context).extension<CakeTextTheme>()!.titleColor,
+                        fontSize: 16,
+                        fontFamily: 'Lato',
+                        fontWeight: FontWeight.bold,
+                        decoration: TextDecoration.none),
+                  ),
+                  onChanged: setSelectedOrderType,
+                  activeColor: Theme.of(context).dividerColor,
```
Comment (OmarHatem28) [inline lib/src/screens/dashboard/widgets/filter_list_widget.dart:100]: let's change this for all radio tiles in this picker to be the same color of the selected item in the `SettingsChoicesCell`

**Review by OmarHatem28 — CHANGES_REQUESTED** [review]

File: macos/Podfile
[the remark this answers is not in the document]
Comment (fossephate) [reply macos/Podfile:1]: this is a fix for building on ios/macos sonoma, I can remove it from the commit, or make a separate PR

[the remark this answers is not in the document]
Comment (OmarHatem28) [reply macos/Podfile:1]: yes please, remove it and make a separate PR for it

**Review by OmarHatem28 — APPROVED** [review]

**Pull request merged**